packages/devtools-proxy-support/src/system-ca.ts

```diff
@@ -31,50 +31,39 @@ function systemCertsCached(systemCAOpts: SystemCAOptions = {}): Promise<{
   return systemCertsCachePromise;
 }
 
+function certToString(cert: string | Uint8Array) {
+  return typeof cert === 'string'
+    ? cert
+    : Buffer.from(cert.buffer, cert.byteOffset, cert.byteLength).toString(
+        'utf8'
+      );
+}
+
 export function mergeCA(...args: (NodeJSCAOption | undefined)[]): string {
   const ca = new Set<string>();
   for (const item of args) {
     if (!item) continue;
-    const caList: readonly (string | Uint8Array)[] = Array.isArray(item)
-      ? item
-      : [item];
+    const caList = Array.isArray(item) ? item : [item];
     for (const cert of caList) {
-      const asString =
-        typeof cert === 'string'
-          ? cert
-          : Buffer.from(cert.buffer, cert.byteOffset, cert.byteLength).toString(
-              'utf8'
-            );
-      ca.add(asString);
+      ca.add(certToString(cert));
     }
   }
   return [...ca].join('\n');
 }
 
-const pemWithParsedCache = new WeakMap<
-  string[],
-  {
-    ca: string[];
-    messages: string[];
-  }
->();
-// TODO(COMPASS-8253): Remove this in favor of OpenSSL's X509_V_FLAG_PARTIAL_CHAIN
-// See linked tickets for details on why we need this (tl;dr: the system certificate
-// store may contain intermediate certficiates without the corresponding trusted root,
-// and OpenSSL does not seem to accept that)
-export function removeCertificatesWithoutIssuer(ca: string[]): {
-  ca: string[];
-  messages: string[];
-} {
-  let result:
-    | {
-        ca: string[];
-        messages: string[];
-      }
-    | undefined = pemWithParsedCache.get(ca);
+export type ParsedX509Cert = { pem: string; parsed: X509Certificate | null };
 
-  const messages: string[] = [];
-  let caWithParsedCerts = ca.map((pem) => {
+/**
+ * Safely parse provided certs, push any encountered errors to the provided
+ * messages array
+ */
+export function parseCACerts(
```

Comment:
Split parsing out of removing function so that I can parse once, then optionally remove, then use the same parsed certs when sorting without loosing the parsed certs inside the "removing" function